public/src/listeners.js

```diff
@@ -0,0 +1,504 @@
+        }
+    });
+
+    window.addEventListener('mouseup', onMouseUp);
+
+    document.getElementById('simulationArea').addEventListener('mousewheel', MouseScroll);
+    document.getElementById('simulationArea').addEventListener('DOMMouseScroll', MouseScroll);
+
+    function MouseScroll(event) {
+        updateCanvasSet(true);
+        event.preventDefault();
+        var deltaY = event.wheelDelta ? event.wheelDelta : -event.detail;
+        event.preventDefault();
+        var deltaY = event.wheelDelta ? event.wheelDelta : -event.detail;
+        const direction = deltaY > 0 ? 1 : -1;
+        handleZoom(direction);
+        updateCanvasSet(true);
+        gridUpdateSet(true);
+
+        if (layoutModeGet())layoutUpdate();
+        else update(); // Schedule update not working, this is INEFFICIENT
+    }
+
+    document.addEventListener('cut', (e) => {
+        if (document.activeElement.tagName == 'INPUT') return;
+
+        if (listenToSimulator) {
+            simulationArea.copyList = simulationArea.multipleObjectSelections.slice();
+            if (simulationArea.lastSelected && simulationArea.lastSelected !== simulationArea.root && !simulationArea.copyList.contains(simulationArea.lastSelected)) {
+                simulationArea.copyList.push(simulationArea.lastSelected);
+            }
+
+
+            var textToPutOnClipboard = copy(simulationArea.copyList, true);
+
+            // Updated restricted elements
+            updateRestrictedElementsInScope();
+
+            localStorage.setItem('clipboardData', textToPutOnClipboard);
+            e.preventDefault();
+            if (textToPutOnClipboard == undefined) return;
+            if (isIe) {
+                window.clipboardData.setData('Text', textToPutOnClipboard);
+            } else {
+                e.clipboardData.setData('text/plain', textToPutOnClipboard);
+            }
+        }
+    });
+
+    document.addEventListener('copy', (e) => {
+        if (document.activeElement.tagName == 'INPUT') return;
+
+        if (listenToSimulator) {
+            simulationArea.copyList = simulationArea.multipleObjectSelections.slice();
+            if (simulationArea.lastSelected && simulationArea.lastSelected !== simulationArea.root && !simulationArea.copyList.contains(simulationArea.lastSelected)) {
+                simulationArea.copyList.push(simulationArea.lastSelected);
+            }
+
+            var textToPutOnClipboard = copy(simulationArea.copyList);
+
+            // Updated restricted elements
+            updateRestrictedElementsInScope();
+
+            localStorage.setItem('clipboardData', textToPutOnClipboard);
+            e.preventDefault();
+            if (textToPutOnClipboard == undefined) return;
+            if (isIe) {
+                window.clipboardData.setData('Text', textToPutOnClipboard);
+            } else {
+                e.clipboardData.setData('text/plain', textToPutOnClipboard);
+            }
```

Comment:
Can we have a wrapping class/function for clipboard management? Noticed sample code repeating.